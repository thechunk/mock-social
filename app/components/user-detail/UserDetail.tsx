import * as React from 'react';
import {PureComponent} from 'react';
import {ScrollView, Text, TouchableWithoutFeedback, View} from 'react-native';
import withUserData from '../../hoc/withUserData';
import withNavigationParamsAsProps from '../../hoc/withNavigationParamsAsProps';
import UserDetailTabs from './UserDetailTabs';
import {NavigationInjectedProps} from 'react-navigation';
import PhotoHero from './PhotoHero';
import styles from './styles';
import UserDetailAlbums from "./UserDetailAlbums";
import {Loader} from "../common/loader/index";
import {inherited} from "../user-list/styles";
import {default as g} from '../../styles/global';

export class UserDetail extends PureComponent<IWithUserDataProps & IWithUserDataOptionalProps &
    IWithNavigationParamsAsProps & NavigationInjectedProps> {
    render() {
        return this.props.loading ? <Loader loading={this.props.loading} style={inherited.loader} /> : (
            <ScrollView contentContainerStyle={styles.container} style={g.viewBackground}>
                {this.props.user
                    ? (
                        <>
                            <View style={styles.photoNameView}>
                                <View style={[g.containerShadow, g.flex1]}>
                                    <View style={[styles.photoHeroContainer, g.roundedBorder]}>
                                        <PhotoHero userId={this.props.user.id} />
                                    </View>
                                </View>
                                <View style={styles.nameOverlay}>
                                    <Text style={styles.title}>{this.props.user.name}</Text>
                                    <Text style={styles.subtitle}>at {this.props.user.company.name}</Text>
                                </View>
                            </View>
                            <View style={[styles.callOutSection, g.roundedBorder]}>
                                <Text style={styles.callOutSectionText}>{this.props.user.company.catchPhrase}</Text>
                            </View>

                            <Text style={styles.sectionHeaderText}>My Address</Text>
                            <View style={[styles.callOutSection, g.roundedBorder]}>
                                <TouchableWithoutFeedback>
                                    <View >
                                        <Text>
                                            {`${this.props.user.address.suite} ${this.props.user.address.street}`}
                                        </Text>
                                        <Text>{this.props.user.address.city}</Text>
                                        <Text>{this.props.user.address.zipcode}</Text>
                                    </View>
                                </TouchableWithoutFeedback>
                            </View>

                            <Text style={styles.sectionHeaderText}>My Albums</Text>
                            <UserDetailAlbums userId={this.props.user.id} />

                            <Text style={styles.sectionHeaderText}>My Activity</Text>
                            <UserDetailTabs user={this.props.user} />
                        </>
                    )
                    : null
                }
            </ScrollView>
        )
    }
}

export default withNavigationParamsAsProps(withUserData(UserDetail));