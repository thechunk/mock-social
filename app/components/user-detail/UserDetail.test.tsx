import * as React from 'react';
import {shallow} from 'enzyme';
import {UserDetail} from './UserDetail';

describe('UserDetail', () => {
    it('renders correctly', () => {
        expect(shallow(<UserDetail loading={false} user={{id: 1, name: "Name", company: {name: "Company", catchPhrase: "catchPhrase"}}} />)).toMatchSnapshot();
    });
});
