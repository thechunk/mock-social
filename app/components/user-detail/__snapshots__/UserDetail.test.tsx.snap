// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetail renders correctly 1`] = `
<ScrollViewMock
  contentContainerStyle={
    Array [
      Object {
        "alignItems": "stretch",
        "flexDirection": "column",
      },
      Object {
        "paddingBottom": 25,
        "paddingTop": 89,
      },
    ]
  }
  scrollIndicatorInsets={
    Object {
      "bottom": 25,
      "top": 89,
    }
  }
  style={
    Object {
      "backgroundColor": "#a6abbd",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingBottom": 12,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "elevation": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 6,
              "width": 0,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 5,
          },
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "margin": 12,
            },
            Object {
              "borderBottomLeftRadius": 12,
              "borderBottomRightRadius": 12,
              "borderTopLeftRadius": 12,
              "borderTopRightRadius": 12,
              "overflow": "hidden",
            },
          ]
        }
      >
        <PhotoHero
          userId={1}
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "flexDirection": "column",
          "height": 180,
          "justifyContent": "center",
          "marginRight": 12,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Name
      </Text>
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 14,
          }
        }
      >
        at 
        Company
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "marginBottom": 12,
          "marginLeft": 12,
          "marginRight": 12,
          "padding": 24,
        },
        Object {
          "borderBottomLeftRadius": 12,
          "borderBottomRightRadius": 12,
          "borderTopLeftRadius": 12,
          "borderTopRightRadius": 12,
          "overflow": "hidden",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 14,
        }
      }
    >
      catchPhrase
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#000",
        "fontWeight": "bold",
        "margin": 12,
      }
    }
  >
    My Address
  </Text>
  <withNavigation()
    user={
      Object {
        "company": Object {
          "catchPhrase": "catchPhrase",
          "name": "Company",
        },
        "id": 1,
        "name": "Name",
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#000",
        "fontWeight": "bold",
        "margin": 12,
      }
    }
  >
    My Albums
  </Text>
  <withNavigation(UserDetailAlbums)
    userId={1}
  />
  <Text
    style={
      Object {
        "color": "#000",
        "fontWeight": "bold",
        "margin": 12,
      }
    }
  >
    My Activity
  </Text>
  <UserDetailTabs
    user={
      Object {
        "company": Object {
          "catchPhrase": "catchPhrase",
          "name": "Company",
        },
        "id": 1,
        "name": "Name",
      }
    }
  />
</ScrollViewMock>
`;
